main.py

```diff
@@ -0,0 +1,319 @@
+            "highest_humidity": highest_humidity,
+            "highest_humidity_day": highest_humidity_day
+        }
+
+    @staticmethod
+    def calculate_monthly_averages(readings, year, month):
+        total_max_temp = 0
+        total_min_temp = 0
+        total_mean_humidity = 0
+        count_max_temp = 0
+        count_min_temp = 0
+        count_mean_humidity = 0
+
+        for reading in readings:
+            reading_year, reading_month = reading.date.split('-')[:2]
+            if reading_year == year and reading_month == month:
+                if reading.max_temp is not None:
+                    total_max_temp += reading.max_temp
+                    count_max_temp += 1
+                if reading.min_temp is not None:
+                    total_min_temp += reading.min_temp
+                    count_min_temp += 1
+                if reading.mean_humidity is not None:
+                    total_mean_humidity += reading.mean_humidity
+                    count_mean_humidity += 1
+
+        avg_max_temp = total_max_temp / count_max_temp if count_max_temp \
+            else None
+        avg_min_temp = total_min_temp / count_min_temp if count_min_temp \
+            else None
+        avg_mean_humidity = total_mean_humidity / count_mean_humidity if (
+            count_mean_humidity) else None
+
+        return {
+            "avg_max_temp": avg_max_temp,
+            "avg_min_temp": avg_min_temp,
+            "avg_mean_humidity": avg_mean_humidity
+        }
+
+
+class WeatherReportsGenerator:
+    def __init__(self, readings):
+        self.readings = readings
+
+    def display_yearly_extremes(self, year):
+        extremes = WeatherResultsCalculator.calculate_yearly_extremes(
+            self.readings, year)
+
+        print(
+            f"Highest: {extremes['highest_temp']}C on "
+            f"{self.format_date(extremes['highest_temp_day'])}")
+        print(
+            f"Lowest: {extremes['lowest_temp']}C on "
+            f"{self.format_date(extremes['lowest_temp_day'])}")
+        print(
+            f"Humidity: {extremes['highest_humidity']}% on "
+            f"{self.format_date(extremes['highest_humidity_day'])}")
+
+    @staticmethod
+    def format_date(date_str):
+        parts = date_str.split('-')
+        # year = parts[0]
+        month = parts[1]
+        day = parts[2]
+        month_name = WeatherReportsGenerator.get_month_name(int(month))
+        return f"{month_name} {day}"
+
+    @staticmethod
+    def get_month_name(month):
+        months = ["January", "February", "March", "April", "May", "June",
+                  "July", "August", "September", "October", "November",
+                  "December"]
+        return months[month - 1]
+
+    def display_monthly_averages(self, year, month):
+        averages = WeatherResultsCalculator.calculate_monthly_averages(
+            self.readings, year, month)
+        print(f"Highest Average: {averages['avg_max_temp']:.2f}C")
```

Comment:
there should be checks to show a different message instead of NoneC in case of None in averages['avg_max_temp']